src/widgets/QDisassemblyView.cpp

```diff
@@ -989,18 +989,25 @@ void QDisassemblyView::paintEvent(QPaintEvent *) {
 			QString immed_string;
 			for (unsigned int op_idx = 0; op_idx < op_count; op_idx++) {
 				auto oper = inst.operands()[op_idx];
-				edb::address_t address = 0;
+				edb::address_t ascii_address = 0;
 				if (oper.type() == edb::Operand::TYPE_REL || oper.type() == edb::Operand::TYPE_IMMEDIATE) {
-					address = oper.relative_target();
-				} else if (oper.type() == edb::Operand::TYPE_EXPRESSION) {
-					// TODO Interact with capstone to evaluate operand's displacement
+					ascii_address = oper.relative_target();
+				} else if (
+					oper.type() == edb::Operand::TYPE_EXPRESSION &&
+					oper.expression().displacement_type == edb::Operand::DISP_PRESENT)
+				{
+					if (oper.expression().base == edb::Operand::Register::X86_REG_RIP) {
+						ascii_address += address + oper.owner()->size() + oper.expression().displacement;
+					} else if (oper.expression().displacement > 0) {
+						ascii_address = oper.expression().displacement;
```

Comment:
Depends on your definition of wrong. :smile: We could substitute the current values in those registers and solve the address,  but unless you are paused on that instruction those results are probably "wrong" as the relevant registers are probably going to be garbage values until the routine fixes them right before the instruction runs. Honestly I was only really worried about the RIP relative and the constant ASCII as they were low-hanging fruit and hadn't thought through this much.